tools/qontract_cli.py

```diff
@@ -1052,6 +1052,62 @@ def clusters_network(ctx, name):
     print_output(ctx.obj["options"], clusters, columns)
 
 
+@get.command()
+@click.pass_context
+def cidr_blocks(ctx) -> None:
+    import ipaddress
+
+    from reconcile.typed_queries.aws_vpcs import get_aws_vpcs
+
+    columns = ["type", "name", "account", "cidr", "from", "to", "hosts", "overlaps"]
+
+    clusters = [c for c in queries.get_clusters() if c.get("network")]
+    cidrs = [
+        {
+            "type": "cluster",
+            "name": c["name"],
+            "account": ((c.get("spec") or {}).get("account") or {}).get("name"),
+            "cidr": c["network"]["vpc"],
+            "from": str(ipaddress.ip_network(c["network"]["vpc"])[0]),
+            "to": str(ipaddress.ip_network(c["network"]["vpc"])[-1]),
+            "hosts": str(ipaddress.ip_network(c["network"]["vpc"]).num_addresses),
+            "description": c.get("description"),
+        }
+        for c in clusters
+    ]
+
+    vpcs = get_aws_vpcs()
+    cidrs.extend(
+        [
+            {
+                "type": "vpc",
+                "name": vpc.name,
+                "account": vpc.account.name,
+                "cidr": vpc.cidr_block,
+                "from": str(ipaddress.ip_network(vpc.cidr_block)[0]),
+                "to": str(ipaddress.ip_network(vpc.cidr_block)[-1]),
+                "hosts": ipaddress.ip_network(vpc.cidr_block).num_addresses,
+                "description": vpc.description,
+            }
+            for vpc in vpcs
+        ]
+    )
+
+    for index, cidr in enumerate(cidrs):
+        network = ipaddress.ip_network(cidr["cidr"])
+        overlaps = [
+            f"{c['type']}/{c['name']}"
+            for i, c in enumerate(cidrs)
+            if i != index and network.overlaps(ipaddress.ip_network(c["cidr"]))
+        ]
+        cidr["overlaps"] = ", ".join(overlaps)
```

Comment:
nit: for X cidrs it is X*X-1 comparisons. considering our low numbe of CIRDs i think this is absolutely ok for now and the report is of great value.

with the rise of AWS account numbers, we need to keep this in mind though

Reply:
I did not put a lot of thoughts into this. is there an easy way to get more efficient?

We could maybe slice the second `enumerate(cidrs)` as `enumerate(cidrs[index+1:])` but would need to register overlaps both ways somehow

Reply:
current run time:
```
real	0m4.123s
user	0m2.544s
sys	0m0.614s
```

Reply:
can use [itertools.combinations](https://docs.python.org/3/library/itertools.html#itertools.combinations) to get all combos, then collect overlaps, then iterate cidrs once to set them.

```python
    overlaps = defaultdict(list)
    for a, b in itertools.combinations(cidrs, 2):
        if ipaddress.ip_network(a["cidr"]).overlaps(ipaddress.ip_network(b["cidr"])):
            key_a = f"{a['type']}/{a['name']}"
            key_b = f"{b['type']}/{b['name']}"
            overlaps[key_a].append(key_b)
            overlaps[key_b].append(key_a)
    for cidr in cidrs:
        cidr["overlaps"] = ", ".join(overlaps[f"{cidr['type']}/{cidr['name']}"])
```

In terms of time complexity, it's still O(N^2), further improvement requires a different way to find overlap, not worth it now.

Reply:
yeah, i am not sure it makes it more readable nor more efficient. So if you're both fine, I'll keep it as is for now

Reply:
yes, keep it as is for now, no big differences considering our data size.

Reply:
Just note here in case later we need optimize this or similar features, a more efficient way is using [Interval Tree](https://en.wikipedia.org/wiki/Interval_tree), O(NlogN).